iexec-sms-library/src/main/java/com/iexec/sms/api/TeeSessionGenerationError.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2022 IEXEC BLOCKCHAIN TECH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.iexec.sms.api;
+
+public enum TeeSessionGenerationError {
+    REQUEST_NOT_SIGNED_BY_HIMSELF,
+
+    EXECUTION_NOT_AUTHORIZED_EMPTY_PARAMS_UNAUTHORIZED,
+    EXECUTION_NOT_AUTHORIZED_NO_MATCH_ONCHAIN_TYPE,
+    EXECUTION_NOT_AUTHORIZED_GET_CHAIN_TASK_FAILED,
+    EXECUTION_NOT_AUTHORIZED_TASK_NOT_ACTIVE,
+    EXECUTION_NOT_AUTHORIZED_GET_CHAIN_DEAL_FAILED,
+    EXECUTION_NOT_AUTHORIZED_INVALID_SIGNATURE,
+
+    GET_TASK_DESCRIPTION_FAILED,
+
+    NO_SESSION_REQUEST,
+    NO_TASK_DESCRIPTION,
```

Comment:
```suggestion
     // generic errors
    NO_TASK_DESCRIPTION,
    ...
```

Reply:
What do you mean with `generic errors`? Is that generic opposed to stage-specific values? In that case, there are other values to put under this label.

Reply:
Yes, other values under this label ( `...`)
If I'm not wrong, some values are thrown in different locations, some are not

Reply:
The only one that's thrown in different locations is `NO_TASK_DESCRIPTION`.
Nonetheless, I have sorted all the enum values in https://github.com/iExecBlockchainComputing/iexec-sms/pull/93/commits/938bb042fe65a174dab9a85d23254b1d47a1a53b, let me know if this is OK for you :)